src/Plugin/GraphQL/DataProducer/Routing/RouteEntity.php

```diff
@@ -107,7 +107,15 @@ public function __construct(
    */
   public function resolve($url, ?string $language, FieldContext $context): ?Deferred {
     if ($url instanceof Url) {
+      // Check that this route is actually an entity route.
+      if (strpos($url->getRouteName(), 'entity.') === FALSE) {
```

Comment:
Maybe we can simplify this by calling `$url->getRouteName()` only once. Like this:

```
[$baseType, $type] = explode('.', $url->getRouteName());

if ($baseType !== 'entity' || empty($type)) {
  return NULL;
}     
```

Reply:
Nice suggestion, added it too.